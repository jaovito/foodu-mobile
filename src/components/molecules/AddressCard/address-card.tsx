import React from 'react';
import {AddressCardProps} from './address-card.props';
import {
  Container,
  Address,
  PostalCode,
  City,
  State,
} from './address-card.styles';

export const AddressCard = ({
  address,
  postalCode,
  city,
  state,
  onPress,
}: AddressCardProps) => {
  return (
    <Container onPress={onPress}>
      <Address>{address}</Address>

      {postalCode && <PostalCode>{postalCode}</PostalCode>}

      {city && <City>{city}</City>}

      {state && <State>{state}</State>}
    </Container>
  );
};
