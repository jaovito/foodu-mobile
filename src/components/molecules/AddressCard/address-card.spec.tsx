import {render, fireEvent} from '@test-utils';
import {mock} from '../../../test/mocks';
import {AddressCard} from './address-card';

describe('AddressCard tests', () => {
  it('should be defined', () => {
    const card = render(
      <AddressCard
        address={mock.addressCard.title}
        city={mock.addressCard.city}
        postalCode={mock.addressCard.postalCode}
        state={mock.addressCard.state}
      />,
    );

    expect(card).toBeDefined();
  });

  it('should be able to find the address text', async () => {
    const {findByText} = render(
      <AddressCard
        address={mock.addressCard.title}
        city={mock.addressCard.city}
        postalCode={mock.addressCard.postalCode}
        state={mock.addressCard.state}
      />,
    );

    const address = await findByText(mock.addressCard.title);

    expect(address).toBeTruthy();
  });

  it('should be able to press in the card', async () => {
    const onPress = jest.fn();
    const {findByText} = render(
      <AddressCard
        address={mock.addressCard.title}
        city={mock.addressCard.city}
        postalCode={mock.addressCard.postalCode}
        state={mock.addressCard.state}
        onPress={onPress}
      />,
    );

    const address = await findByText(mock.addressCard.title);
    fireEvent.press(address);

    expect(onPress).toHaveBeenCalled();
  });

  it('should match with snapshot without onPress', async () => {
    const card = render(
      <AddressCard
        address={mock.addressCard.title}
        city={mock.addressCard.city}
        postalCode={mock.addressCard.postalCode}
        state={mock.addressCard.state}
      />,
    ).toJSON();

    expect(card).toMatchSnapshot();
  });

  it('should match with snapshot with onPress', async () => {
    const onPress = jest.fn();
    const card = render(
      <AddressCard
        address={mock.addressCard.title}
        city={mock.addressCard.city}
        postalCode={mock.addressCard.postalCode}
        state={mock.addressCard.state}
        onPress={onPress}
      />,
    ).toJSON();

    expect(card).toMatchSnapshot();
  });
});
