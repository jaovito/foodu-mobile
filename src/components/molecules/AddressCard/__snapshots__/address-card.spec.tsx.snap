// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCard tests should match with snapshot with onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "marginBottom": 31,
      "opacity": 1,
      "paddingBottom": 31,
      "paddingLeft": 31,
      "paddingRight": 31,
      "paddingTop": 31,
    }
  }
>
  <Text
    fontWeight="semiBold"
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_600SemiBold",
          "fontSize": 31,
        },
        {
          "fontSize": 31,
          "marginBottom": 31,
        },
      ]
    }
  >
    Some address title
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    12345
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    San Francisco
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    California
  </Text>
</View>
`;

exports[`AddressCard tests should match with snapshot without onPress 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 8,
      "marginBottom": 31,
      "opacity": 1,
      "paddingBottom": 31,
      "paddingLeft": 31,
      "paddingRight": 31,
      "paddingTop": 31,
    }
  }
>
  <Text
    fontWeight="semiBold"
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_600SemiBold",
          "fontSize": 31,
        },
        {
          "fontSize": 31,
          "marginBottom": 31,
        },
      ]
    }
  >
    Some address title
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    12345
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    San Francisco
  </Text>
  <Text
    style={
      [
        {
          "color": "#0D0D0D",
          "fontFamily": "Poppins_400Regular",
          "fontSize": 31,
        },
        {
          "fontSize": 24,
        },
      ]
    }
  >
    California
  </Text>
</View>
`;
